scripts/west_commands/runners/jlink.py

```diff
@@ -25,7 +26,31 @@
 except ImportError:
     MISSING_REQUIREMENTS = True
 
-DEFAULT_JLINK_EXE = 'JLink.exe' if sys.platform == 'win32' else 'JLinkExe'
+if sys.platform == 'win32':
+    # JLink.exe can collide with the JDK executable of the same name
+    # Look in the usual locations before falling back to $PATH
+    for root in [os.environ["ProgramFiles"], os.environ["ProgramFiles(x86)"]]:
```

Comment:
The `SIM112` appears to be a false positive, you can add `# noqa SIM112` to ignore it.